header.php

```diff
@@ -4,14 +4,17 @@
 if (!empty($_SESSION['$user_id'])) {
     $_SESSION['logged_in'] = true;
     $_SESSION['is_company'] = false;
-    $result_company = mysqli_query($db_connect, "SELECT company_name FROM users WHERE id= " . $_SESSION['$user_id'] . " ");
-    $matches_company = mysqli_num_rows($result_company);
-    if (0 !== $matches_company) {
-        $rows = $result_company->fetch_assoc();
-        if ($rows['company_name'] != "") {
-            $_SESSION['is_company'] = true;
-        }
-    }
+
+    $query = "SELECT company_name FROM users WHERE id=?"; // SQL with parameters
+    $stmt = $db_connect->prepare($query);
+    $stmt->bind_param("i", $_SESSION['$user_id']);
+    $stmt->execute();
+    $result = $stmt->get_result(); // get the mysqli result
+    $company_name_fetched = $result->fetch_assoc(); // fetch data
```

Comment:
Hmmm, you should check if `$stmt->get_result()` actually returned anything.

`get_result()` should return either the results object or false. so checking if `$result` is not false on the line below should be sufficient.